### PR TITLE
Centralize request error handling

### LINKED ISSUE #2655: Centralize request error handling
There are lots of `try..catch` blocks in our API routes. We can simplify the codebase with error handling middleware.

### This task is done when...
 - [ ] error handling middleware is added to the API

### PR DESCRIPTION
**Pull request by radavis**
Resolves #2655

### This pull request...

- adds error-handling middleware to the api
- adds mockResponse and mockExpress test helpers
- changes calls to `logger.silly` changed to `logger.debug`

### This pull request is ready to merge when...

- [x] Tests have been updated (and all tests are passing)
- [x] This code has been reviewed by someone other than the original author
- [x] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented
- [x] The change has been documented
  - [x] Associated OpenAPI documentation has been updated
- [x] Changelog is updated as appropriate

### This feature is done when...

- [ ] ~Design has approved the experience~
- [ ] ~Product has approved the experience~

No changes to UI or app functionality. Just cleaning up the codebase a little.

### CODE REVIEW

**Review by thetif — COMMENTED** [review]
Nice job! I like how much more streamlined and standardized the error messages are this way.

I did see that async/await and then/catch were being used together, you only need one or the other. 

https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await

The documentation above says you can use `await function.catch()` but I've never used it that way before so I don't know for sure about that.

**Comment by radavis** [comment]
Thanks for the feedback, Tif!

> I did see that async/await and then/catch were being used together, you only need one or the other.

Could you point a specific example of this?

File: api/routes/apds/delete.js [hunk 1]
```diff
@@ -3,20 +3,16 @@ const { can, userCanEditAPD } = require('../../middleware');
...
-    async (req, res) => {
-      try {
-        await deleteAPDByID(req.meta.apd.id);
-        res.status(204).end();
-      } catch (e) {
-        logger.error({ id: req.id, message: e });
-        res.status(500).end();
-      }
+    async (req, res, next) => {
+      await deleteAPDByID(req.meta.apd.id)
```
Comment (thetif) [inline api/routes/apds/delete.js:13]: await and then

File: api/routes/auth/activities/index.js [hunk 2]
```diff
@@ -1,23 +1,10 @@
...
-          .map(({ name }) => name)
-          .join(', ')}`
-      );
-      res.send(activities);
-    } catch (e) {
-      logger.error({ id: req.id, message: e });
-      res.status(500).end();
-    }
+  app.get('/auth/activities', can('view-roles'), async (req, res, next) => {
+    await getAuthActivities()
```
Comment (thetif) [inline api/routes/auth/activities/index.js:6]: await then

File: api/routes/auth/roles/get.js [hunk 3]
```diff
@@ -1,22 +1,10 @@
...
-        id: req.id,
-        message: `got all the roles: ${roles.map(({ name }) => name).join(', ')}`
-      });
-      res.send(roles);
-    } catch (e) {
-      logger.error({ id: req.id, message: e });
-      res.status(500).end();
-    }
+  app.get('/auth/roles', can('view-roles'), async (req, res, next) => {
+    await getActiveAuthRoles()
```
Comment (thetif) [inline api/routes/auth/roles/get.js:6]: await then

File: api/routes/states/get.js [hunk 4]
```diff
@@ -2,12 +2,13 @@ const { raw: knex } = require('../../db');
 const { loggedIn } = require('../../middleware/auth');
 
 module.exports = (app) => {
-  app.get('/states', async (request, response) => {
+  app.get('/states', async (request, response, next) => {
     await knex('states').select('id', 'name')
```
Comment (thetif) [inline api/routes/states/get.js:6]: await then

File: api/routes/users/get.js [hunk 5]
```diff
@@ -1,37 +1,19 @@
...
-    try {
-      const users = await getAllUsers();
-      logger.silly({ id: req.id, message: 'sending users' });
-      res.send(users);
-    } catch (e) {
-      logger.error({ id: req.id, message: e });
-      res.status(500).end();
-    }
+  app.get('/users', can('view-users'), async (req, res, next) => {
+    await getAllUsers()
```
Comment (thetif) [inline api/routes/users/get.js:6]: await then

File: api/routes/users/get.js [hunk 6]
```diff
@@ -1,37 +1,19 @@
...
-      } else {
-        logger.verbose({ id: req.id, message: `no user found [${req.params.id}]`});
-        res.status(404).end();
-      }
-    } catch (e) {
-      logger.error({ id: req.id, message: e });
-      res.status(500).end();
-    }
+  app.get('/users/:id', can('view-users'), async (req, res, next) => {
+    await getUserByID(req.params.id)
```
Comment (thetif) [inline api/routes/users/get.js:12]: await then

**Review by thetif — CHANGES_REQUESTED** [review]

File: api/routes/users/get.js (hunk 6 above)
Comment (radavis) [reply api/routes/users/get.js:12]: Ah, I see. I was sprinkling async/await everywhere when it isn't necessary.

Comment (thetif) [reply api/routes/users/get.js:12]: you just wanted to make sure it really waited until it was finished :D

Comment (radavis) [reply api/routes/users/get.js]: Unfortunately, the unit tests fail without async/await present

**Review by thetif — APPROVED** [review]

**Pull request merged**